### ISSUE TITLE
SC2206: mention pathname expansion in the warning message

### ISSUE THREAD

**Issue by Flupp** [koalaman/shellcheck#1626]
#### For bugs
- Rule Id: SC2206
- My shellcheck version (`shellcheck --version` or "online"): 0.6.0 and online
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2206)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
a=($x)
printf '%s\n' "${a[@]}"
```

#### Here's what shellcheck currently says:

```
Line 2:
a=($x)
   ^-- SC2206: Quote to prevent word splitting, or split robustly with mapfile or read -a.
```

#### Here's what I wanted or expected to see:

```
Line 2:
a=($x)
   ^-- SC2206: Quote to prevent word splitting and pathname expansion, or split robustly with mapfile or read -a.
```
I think, [SC2206](https://shellcheck.net/wiki/SC2206) should mention pathname expansion in the warning message and in the page title in the wiki.

**Comment by koalaman** [comment]
I can see this avoiding some confusion for people who *do* want to glob. I've updated the message, thanks!
